### PR TITLE
Remove timeout from add2app test for iOS

### PR DESCRIPTION
**Pull request by dnfield**
## Description

Fixes the flake in ios_add2app and removes the timeout.  We still have retry logic in here - eventually, EarlGrey will be able to do this without us needing this logic.

This PR will fail until the related engine PR lands in the framework.

## Related Issues

https://github.com/flutter/engine/pull/8005
https://github.com/flutter/flutter/issues/24519 (related, but for something slightly different).

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes tests for *all* changed/updated/fixed behaviors (See [Test Coverage]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by dnfield** [comment]
Upstream change has landed, will just need to get it rolled into framework

**Comment by dnfield** [comment]
Once https://github.com/flutter/flutter/pull/28688 this can be rebased and should work.

File: dev/integration_tests/ios_add2app/ios_add2appTests/IntegrationTests.m [hunk 1]
```diff
@@ -6,12 +6,30 @@
...
+  __block id<NSObject> observer = [[NSNotificationCenter defaultCenter]
+      addObserverForName:@"FlutterSemanticsUpdate"
+                  object:viewController
+                   queue:nil
+              usingBlock:^(NSNotification *notification) {
+                semanticsAvailable = YES;
+                [[NSNotificationCenter defaultCenter] removeObserver:observer];
+              }];
+  [viewController.engine ensureSemanticsEnabled];
+  while (semanticsAvailable == NO && tries != 0) {
```
Comment (xster) [inline dev/integration_tests/ios_add2app/ios_add2appTests/IntegrationTests.m]: Add one line saying what is synchronous and what isn't in a comment here and in the ensureSemanticsEnabled objective-c doc

**Review by xster — COMMENTED** [review]
LGTM

**Pull request merged**